lib/vrl/vrl/benches/vm.rs

```diff
@@ -0,0 +1,67 @@
+use std::collections::BTreeMap;
+
+use compiler::state;
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use indoc::indoc;
+use vector_common::TimeZone;
+use vrl::{Runtime, Value};
+
+static SOURCE: [(&str, &str); 2] = [
+    (
+        "parse_json",
+        indoc! {r#"
+            x = parse_json!(s'{"noog": "nork"}')
+            x.noog
+        "#},
+    ),
+    (
+        "simple",
+        indoc! {r#"
+            .hostname = "vector"
+
+            if .status == "warning" {
+            .thing = upcase(.hostname)
+            } else if .status == "notice" {
+            .thung = downcase(.hostname)
+            } else {
+            .nong = upcase(.hostname)
+            }
+
+            .matches = { "name": .message, "num": "2" }
+            .origin, .err = .hostname + "/" + .matches.name + "/" + .matches.num
+        "#},
+    ),
+];
+
+fn benchmark_kind_display(c: &mut Criterion) {
+    let mut group = c.benchmark_group("vrl_compiler/value::kind::display");
+    for (name, source) in &SOURCE {
+        let state = state::Runtime::default();
+        let runtime = Runtime::new(state);
+        let tz = TimeZone::default();
+        let functions = vrl_stdlib::all();
+        let program = vrl::compile(source, &functions, None).unwrap();
+        let vm = runtime.compile(functions, &program).unwrap();
+
+        group.bench_with_input(BenchmarkId::new("Vm", name), &vm, |b, vm| {
+            b.iter(|| {
+                let state = state::Runtime::default();
+                let mut runtime = Runtime::new(state);
```

Comment:
Just noting that Stephen and I discussed moving these setup steps out into `iter_with_setup` to avoid them being included in the benchmarks.

Reply:
We could move the `Value::Object(BTreeMap::default())` up into an input too.

Reply:
Yeah, was also thinking about that. However, I think they should include at least resetting the state - which is what we do in real usage.

Reply:
Yeah, my reading of these benchmarks is that we are explicitly interested in the runtime performance, especially given the setup steps are the same for both the VM and tree-walking.

Reply:
@jszwedko just noting that creating and resetting the runtime will be necessary for traverse and VM, but not for the LLVM approach. For which reason I'd keep them in, at least when we add the LLVM approach to the comparison.

Reply:
It seems like creating the runtime only happens once though, at Vector boot? In that case, I think we are more interested in the event throughput after the initial setup. If we were doing the setup for each event, then I would agree.

Reply:
Creating the runtime, yes. Resetting it needs to be done at each iteration, though.

Reply:
Ah, yeah, I see it. Then yes, I think we should keep the `runtime.clear()` in the benchmark since that is done for each event.